app/delegate-server/Lib/AVar.purs

```diff
@@ -5,17 +5,17 @@ module DelegateServer.Lib.AVar
 
 import Prelude
 
-import Effect.Aff (Aff)
 import Effect.Aff.AVar (AVar)
 import Effect.Aff.AVar (put, take) as AVar
+import Effect.Aff.Class (class MonadAff, liftAff)
 
-modifyAVar :: forall (a :: Type). AVar a -> (a -> Aff a) -> Aff a
+modifyAVar :: forall m a. MonadAff m => AVar a -> (a -> m a) -> m a
 modifyAVar avar f = do
-  val <- f =<< AVar.take avar
-  AVar.put val avar
+  val <- f =<< liftAff (AVar.take avar)
+  liftAff $ AVar.put val avar
```

Comment:
Can it block here or not?

Reply:
These functions are not deadlock-safe and should be used with caution, namely, they should only be applied to AVars that are not known to be empty or that are expected to be filled.